GomokuServer/src/GomokurServer.Core/Games/Entities/Game.cs

```diff
@@ -215,7 +215,7 @@ public virtual RematchResult Rematch(string playerId)
 			return canRematchResult;
 		}
 
-		var newGame = new Game(GameSettings, new Players(Players.White, Players.Black), _dateTimeProvider)
+		var newGame = new Game(GameSettings, Players.GetSwitchedColors(), _dateTimeProvider)
```

Comment:
Maybe, also makes sense to add validation in the GetSwitchedColors() method to ensure both players exist before swapping colors

Reply:
One thing what I didn't liked in previous solution, that players were nullable type, while they can be null only when Game is waiting for players. I did not touched this, unless I started automatic players lookup feature. That feature showed me one more time, that my solution was suboptimal. So, basically now Players is invariant for Game class. If game instance is created, Players always there